src/main/java/com/exactpro/epfast/decoder/presencemap/DecodePresenceMap.java

```diff
@@ -0,0 +1,80 @@
+package com.exactpro.epfast.decoder.presencemap;
+
+import com.exactpro.epfast.decoder.IDecodeContext;
+import io.netty.buffer.ByteBuf;
+
+import java.util.BitSet;
+
+public class DecodePresenceMap implements IDecodeContext {
+
+    private BitSet value = new BitSet();
+
+    private int truncationIndex = 0;
+
+    private int setIndex = 0;
+
+    private boolean ready;
+
+    private boolean overlong;
+
+    public void decode(ByteBuf buf) {
+        reset();
+        int readerIndex = buf.readerIndex();
+        int readLimit = buf.writerIndex();
+        while ((readerIndex < readLimit) && !ready) {
+            accumulateValue(buf.getByte(readerIndex++));
+        }
+        checkOverlongAndTruncate();
+        buf.readerIndex(readerIndex);
+    }
+
+    public void continueDecode(ByteBuf buf) {
+        int readerIndex = buf.readerIndex();
+        int readLimit = buf.writerIndex();
+        while ((readerIndex < readLimit) && !ready) {
+            accumulateValue(buf.getByte(readerIndex++));
+        }
+        checkOverlongAndTruncate();
+        buf.readerIndex(readerIndex);
+    }
+
+    public boolean getValue(int index) {
+        return value.get(index);
+    }
+
+    public boolean isReady() {
+        return ready;
+    }
+
+    public boolean isOverlong() {
+        return overlong;
+    }
+
+    private void accumulateValue(int oneByte) {
+        if (oneByte < 0) { // if stop bit is set
+            oneByte &= CLEAR_STOP_BIT_MASK;
+            ready = true;
+        }
+        for (int i = 0; i < 7; i++) {
+            value.set(setIndex++, (((oneByte >> i) & 0b00000001) != 0));
+        }
+        if (oneByte != 0) {
+            truncationIndex = setIndex;
+        }
+    }
+
+    public final void reset() {
+        value = new BitSet();
+        ready = false;
+        overlong = false;
+    }
+
+    private void checkOverlongAndTruncate() {
+        if (setIndex > truncationIndex) {
+            value = value.get(0, truncationIndex);
```

Comment:
Does it produces anything different? I think it'll just remove extra zero bytes at the and, but do we actually care?

Reply:
In this implementation it was useless but now thought it may reduce payload when we pass clone of value to PresenceMap class constructor. if it's not important I can remove it

Reply:
`get()` is costly operation even if we forget that new `BitSet` instance is created